diff --color -ur a/gdk-pixbuf/meson.build b/gdk-pixbuf/meson.build
--- a/gdk-pixbuf/meson.build	2021-07-29 16:39:31.504122083 +0200
+++ b/gdk-pixbuf/meson.build	2021-07-29 16:52:11.994364444 +0200
@@ -278,33 +278,35 @@
 dynamic_loaders = []
 dynamic_loaders_dep = []
 
-foreach name, loader: loaders
-  sources = loader.get('sources', [])
-  cond = loader.get('enabled', true)
-  cflags = loader.get('c_args', [])
-  name_suffix = []
+if not meson.is_cross_build()
+  foreach name, loader: loaders
+	sources = loader.get('sources', [])
+	cond = loader.get('enabled', true)
+	cflags = loader.get('c_args', [])
+	name_suffix = []
 
-  # FIXME: workaround for https://gitlab.gnome.org/GNOME/glib/issues/1413
-  if host_system == 'darwin'
-    name_suffix = 'so'
-  endif
+	# FIXME: workaround for https://gitlab.gnome.org/GNOME/glib/issues/1413
+	if host_system == 'darwin'
+	  name_suffix = 'so'
+	endif
 
-  if cond and not (builtin_loaders.contains(name) or builtin_all_loaders)
-    mod = shared_module('pixbufloader-@0@'.format(name),
-                        sources,
-                        gdkpixbuf_enum_h,
-                        dependencies: loaders_deps + gdk_pixbuf_deps + [ gdkpixbuf_dep ],
-                        include_directories: [ root_inc, gdk_pixbuf_inc ],
-                        c_args: common_cflags + gdk_pixbuf_cflags + cflags,
-                        name_suffix: name_suffix,
-                        install: true,
-                        install_dir: gdk_pixbuf_loaderdir)
+	if cond and not (builtin_loaders.contains(name) or builtin_all_loaders)
+	  mod = shared_module('pixbufloader-@0@'.format(name),
+						  sources,
+						  gdkpixbuf_enum_h,
+						  dependencies: loaders_deps + gdk_pixbuf_deps + [ gdkpixbuf_dep ],
+						  include_directories: [ root_inc, gdk_pixbuf_inc ],
+						  c_args: common_cflags + gdk_pixbuf_cflags + cflags,
+						  name_suffix: name_suffix,
+						  install: true,
+						  install_dir: gdk_pixbuf_loaderdir)
 
-    # We need the path to build loaders.cache for tests
-    dynamic_loaders += mod.full_path()
-    dynamic_loaders_dep += mod
-  endif
-endforeach
+	  # We need the path to build loaders.cache for tests
+	  dynamic_loaders += mod.full_path()
+	  dynamic_loaders_dep += mod
+	endif
+  endforeach
+endif
 
 # Build the loaders using native Windows components as dynamic modules, if requested
 if native_windows_loaders
diff --color -ur a/meson.build b/meson.build
--- a/meson.build	2021-07-29 16:39:31.506122069 +0200
+++ b/meson.build	2021-07-29 16:40:20.079831745 +0200
@@ -65,7 +65,7 @@
 gmodule_dep = dependency('gmodule-no-export-2.0', version: glib_req_version,
                          fallback : ['glib', 'libgmodule_dep'])
 gio_dep = dependency('gio-2.0', version: glib_req_version,
-                     fallback : ['glib', 'libgio_dep'])
+                     fallback : ['glib', 'libgio_dep'], static: true)
 
 # Configurations
 gdk_pixbuf_conf = configuration_data()
