diff --color -ur a/configure.ac b/configure.ac
--- a/configure.ac	2021-07-30 13:51:47.371348333 +0200
+++ b/configure.ac	2021-07-30 14:51:55.712706070 +0200
@@ -17,7 +17,7 @@
 AM_CONDITIONAL([BE_WINDOWS], [test x$OS = xWindows_NT])
 AM_COND_IF([BE_WINDOWS], [AC_CONFIG_LINKS([icons/tg-timer.ico:icons/tg-timer.ico])])
 
-CC_CHECK_LDFLAGS([-Wl,--as-needed], [AC_SUBST([AM_LDFLAGS], [-Wl,--as-needed])], [])
+# CC_CHECK_LDFLAGS([-Wl,--as-needed], [AC_SUBST([AM_LDFLAGS], [-Wl,--as-needed])], [])
 CC_CHECK_CFLAGS_APPEND([-Wall -Wextra], [], [])
 
 AC_OUTPUT([Makefile icons/Makefile])
